src/object.h

```diff
@@ -28,4 +30,36 @@ int git_oid__parse(git_oid *oid, const char **buffer_out, const char *buffer_end
 
 void git_oid__writebuf(git_buf *buf, const char *header, const git_oid *oid);
 
+GIT_INLINE(bool) git_object__is_valid(
+	git_repository *repo, const git_oid *id, git_otype type)
+{
+	git_object *obj = NULL;
+	bool valid = true;
+
+	if (git_object__strict_input_validation) {
+		if (git_object_lookup(&obj, repo, id, type) < 0)
```

Comment:
Yes, that's true, but what would be really nice is to have an `object_lookup_type` or something that would use the cached object if it was in the cache and then did a `odb_read_header` if it wasn't.  One sec.